discrete_math/euclidean_algorithm/euclidian.hpp

```diff
@@ -0,0 +1,195 @@
+// USE C++20 MODULES
+// READ e.g. https://learn.microsoft.com/en-us/cpp/cpp/modules-cpp?view=msvc-170
+
+//libraries
+#include <concepts>
+#include <tuple>
+#include <string>
+#include <algorithm>
+#include <numeric>
+#include <ranges>
+#include <iostream>
+#include <print>
+#include <sstream>
+
+namespace algorithms
+{
+	template <std::integral... Is>
+	// Structure containing coefficients and greatest common divisor
+	struct euclidean_result 
+	{
+		// Coefficients
+		std::tuple<Is...> coefficients{}; 
+		// Greatest common divisor
+		std::uint64_t GCD{}; 
+	};
+
+	template <std::integral I, std::integral J>
+	// Function calculating coefficients and gcd for 2 elements
+	constexpr auto gcd_extended(I i, J j) -> euclidean_result<I, J> 
+	{
+		euclidean_result<I, J> result{};
+		// Defining primary coefficients
+		int x = 0, y = 1, x_prev = 1, y_prev = 0;  
+		// Number given for calculation
+		auto a{ i }; 
+		// Number given for calculation
+		auto b{ j };
+		while (b != 0)
+		{
+			// Checking how many times b will fit into a
+			int quotient = a / b; 
+			std::println("a / b");
+			std::println("Dividing number a by number b, result is {} \n", quotient);
+
+			int temp = b;
+			// Counting the rest from division
+			b = a % b; 
+			std::println("b = a % b");
+			std::println("Then we count modulo (rest from division) numbers a and b, write it as number b ");
+			// Switching up places of the variables a and b
+			a = temp; 
+			std::println("Switching up places of a and b, which were counted above");
+
+			int temp_x = x;
+			// Calculating the coefficient of the first given number
+			x = x_prev - quotient * x; 
+			std::println("We calculate coefficient x by subtraction from last coefficient's x product");
+			x_prev = temp_x;
+			std::println("Switching up places of previous coefficient x with temporary coefficient x, which were counted above");
+			std::println("We do the same with coefficient y");
+
+			int temp_y = y;
+			// Calculating the coefficient of the second given number
+			y = y_prev - quotient * y; 
+			y_prev = temp_y;
+		}

```

Comment:
as i mentioned this function should not produce or print any strings.... A possible solution is as follows - return a sequence (std::vector) of numbers produced by this while loop  and write another function `to_description(std::vector)` which produces description given these numbers from the loop.